flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaWriter.java

```diff
@@ -104,9 +103,10 @@ class KafkaWriter<IN>
     // producer pool only used for exactly once
     private final Deque<FlinkKafkaInternalProducer<byte[], byte[]>> producerPool =
             new ArrayDeque<>();
-    private final Closer closer = Closer.create();
     private long lastCheckpointId;
 
+    private final Set<AutoCloseable> closeables = new HashSet<>();
```

Comment:
Lets rename this to maybe `producerCloseables` to be more exact. Will make understanding the `close()` method a bit easier.

Reply:
Also, should this be a `Queue` / `List` instead? We probably never want actual implementation of the producer's equality check to interfere here and leave room to potentially have a producer left unclosed.

Reply:
I renamed the variable to `producerCloseables` and changed its type to `Deque` in [dfcfdb4](https://github.com/apache/flink-connector-kafka/pull/46/commits/dfcfdb47fabf63632edac94139f9acb3c36eb3b5). I chose `Deque` because this is the type used by [Closer](https://github.com/google/guava/blob/3dc3312423209098720e638ac3cb6884fbe77b75/guava/src/com/google/common/io/Closer.java#L111).